python/lsst/sconsUtils/tests.py

```diff
@@ -191,3 +195,68 @@ def run(self, fileGlob):
             self._env.Clean(target, self._tmpDir)
 
         return targets
+
+    def runPythonTests(self, pyList):
+        """Add a single target for testing all python files.
+        pyList is a list of nodes corresponding to python test files.
+        The IgnoreList is respected when scanning for entries.
+        If pyList is None, we will use automated test discovery within
+        pytest. Returns a list containing a single target."""
+
+        if pyList is None:
+            pyList = []
+
+        # Determine any library load path values that we have to prepend
+        # to the command.
+        libpathstr = utils.libraryLoaderEnvironment()
+
+        # Get list of python files with the path included.
+        pythonTestFiles = []
+        for fileGlob in pyList:
+            if not isinstance(fileGlob, basestring):  # env.Glob() returns an scons Node
+                fileGlob = str(fileGlob)
+            for f in glob.glob(fileGlob):
+                if self.ignore(f):
+                    continue
+                pythonTestFiles.append(os.path.join(self._cwd, f))
+
+        # Now set up the python testing target
+        # We always want to run this with the tests target.
+        # We have decided to use pytest caching so that on reruns we only
+        # run failed tests.
+        interpreter = "pytest -Wd --lf --junit-xml=${TARGET} --session2file=${TARGET}.out"
+        target = os.path.join(self._tmpDir, "pytest-{}.xml".format(self._env['eupsProduct']))
+
+        # Work out how many jobs scons has been configured to use
+        # and use that number with pytest. This could cause trouble
+        # if there are lots of binary tests to run and lots of singles.
+        njobs = self._env.GetOption("num_jobs")
+        print("Running pytest with {} processes".format(njobs))
+        if njobs > 1:
+            interpreter = interpreter + " -n {}".format(njobs)
+
+        # Remove target so that we always trigger pytest
+        if os.path.exists(target):
+            os.unlink(target)
+
+        if not pythonTestFiles:
+            print("pytest: automated test discovery mode enabled.")
+        else:
+            nfiles = len(pythonTestFiles)
+            print("pytest: running on {} Python test file{}.".format(nfiles, "" if nfiles == 1 else "s"))
+
+        result = self._env.Command(target, None, """
+        @rm -f ${{TARGET}} ${{TARGET}}.failed;
+        @printf "%s\\n" 'running global pytest... ';
+        @if {2} TRAVIS=1 {0} {1}; then \
+            echo "Global pytest run completed successfully"; \
+        else \
+            echo "Global pytest run: failed"; \
+            mv ${{TARGET}}.out ${{TARGET}}.failed; \
+        fi;
+        """.format(interpreter, " ".join([pipes.quote(p) for p in pythonTestFiles]), libpathstr))
```

Comment:
I had no idea that `pipes.quote` was deprecated. I'm happy to switch to `shlex.quote` next year when we drop python2... We use `pipes.quote` all over the place in `sconsUtils` so making a special note for this one is not going to be helpful.